resources/assets/js/components/ApplicantProfile/Skills/SkillSearchBar.tsx

```diff
@@ -0,0 +1,121 @@
+    description: "The placeholder displayed for the search input field.",
+  },
+  searchButtonText: {
+    id: "searchBar.search.buttonText",
+    defaultMessage: "Search",
+    description: "The placeholder displayed for the search button text.",
+  },
+});
+
+interface SearchBarProps {
+  inputTitle: string;
+  handleSubmit: (locale: string, search: string) => Promise<Skill[]>;
+  handleAddSkill: (skillId: number) => Promise<Skill>;
+}
+
+interface SearchBarValues {
+  locale: string;
+  search: string;
+}
+
+export const SearchBar: React.FunctionComponent<SearchBarProps> = ({
+  inputTitle,
+  handleSubmit,
+  handleAddSkill,
+}: SearchBarProps): React.ReactElement => {
+  const intl = useIntl();
+  const locale = getLocale(intl.locale);
+  const initialValues: SearchBarValues = {
+    locale,
+    search: "",
+  };
+  const validationSchema = Yup.object().shape({
+    search: Yup.string()
+      .required(intl.formatMessage(validationMessages.required))
+      .min(2),
+  });
+
+  const [results, setResults] = React.useState<Skill[]>([]);
+
+  return (
+    <section>
+      <Formik
+        initialValues={initialValues}
+        validationSchema={validationSchema}
+        onSubmit={async (values, { setStatus, setSubmitting }) => {
+          await handleSubmit(values.locale, values.search)
+            .then((searchMatches) => {
+              setStatus("submitted");
+              setResults(searchMatches);
+              setSubmitting(false);
+            })
+            .catch(() => {
+              setSubmitting(false);
+            });
+        }}
+      >
+        {({ status, isSubmitting }): React.ReactElement => (
+          <Form>
+            <div data-h2-grid="b(middle, contained, padded, .25)">
+              <div data-h2-grid-item="b(2of3)">
+                <div data-h2-grid-content>
+                  <FastField
+                    id="search_form_input"
+                    type="search"
+                    name="search"
+                    component={TextInput}
+                    required
+                    label={intl.formatMessage(formMessages.searchLabel)}
+                    placeholder={intl.formatMessage(
+                      formMessages.searchPlaceholder,
+                    )}
+                  />
+                </div>
+              </div>
+              <div data-h2-grid-item="b(1of3)">
+                <div data-h2-grid-content data-h2-display="b(grid)">
+                  <button
+                    type="submit"
+                    disabled={isSubmitting}
+                    data-h2-button="theme-1, round, medium, solid"
+                  >
+                    <span data-h2-button-label>
+                      {inputTitle ||
+                        intl.formatMessage(formMessages.searchButtonText)}
+                    </span>
+                  </button>
+                </div>
+              </div>
+            </div>
+            <SkillSearchResults
+              status={status}
```

Comment:
I don't think it's necessary to pass status down to the SkillSearchResults. Instead, you can conditionally not render it at all until we've submitted a search:
```
{status === "submitted" && (
  <SkillSearchResults ... />
  )}

Reply:
PS: smart use of Formik's setStatus()! I didn't even know about this function. 😄